travis.sh

```diff
@@ -43,14 +43,28 @@ function error {
 
 trap error ERR
 
-git branch --all
-if [ "`git diff origin/master FETCH_HEAD .travis`" != "" ] ; then
-  DIFF=`git diff origin/master FETCH_HEAD .travis | grep .*Subproject | sed s'@.*Subproject commit @@' | sed 'N;s/\n/.../'`;
-  (cd .travis/;git log --oneline --graph --left-right --first-parent --decorate $DIFF) | tee /tmp/$$-travis-diff.log;
-  grep -c '<' /tmp/$$-travis-diff.log && exit 1;
-  echo "ok";
+
+travis_time_start is_jsk_travis_upgraded
+
+# Check if jsk_travis is upgraded, because downgrading jsk_travis is not supported.
+if [ "`git diff origin/master HEAD .travis`" != "" ] ; then
```

Comment:
If I remember this correctly, when travis? jenkins? checkout PR, it
download FETCH_HEAD,

--
◉ Kei Okada


On Tue, Aug 30, 2016 at 3:16 PM, Kentaro Wada <notifications@github.com>
wrote:

> In travis.sh
> <https://github.com/jsk-ros-pkg/jsk_travis/pull/306#discussion_r76735750>:
>
> > @@ -43,14 +43,28 @@ function error {
> >
> >  trap error ERR
> >
> > -git branch --all
> > -if [ "`git diff origin/master FETCH_HEAD .travis`" != "" ] ; then
> > -  DIFF=`git diff origin/master FETCH_HEAD .travis | grep .*Subproject | sed s'@.*Subproject commit @@' | sed 'N;s/\n/.../'`;
> > -  (cd .travis/;git log --oneline --graph --left-right --first-parent --decorate $DIFF) | tee /tmp/$$-travis-diff.log;
> > -  grep -c '<' /tmp/$$-travis-diff.log && exit 1;
> > -  echo "ok";
> > +
> > +travis_time_start is_jsk_travis_upgraded
> > +
> > +# Check if jsk_travis is upgraded, because downgrading jsk_travis is not supported.
> > +if [ "`git diff origin/master HEAD .travis`" != "" ] ; then
>
> @k-okada <https://github.com/k-okada> I think the diff should be between
> origin/master and HEAD. Why did you set FETCH_HEAD?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_travis/pull/306/files/81d14542fec40fa2e680268b96dba546b1bec369#r76735750>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3HD9Z8NUFy4ZqUFMwChytDrehoadks5qk8q_gaJpZM4JwLYz>
> .
>